docs/tutorials/minimal-working-example-react.mdx

```diff
@@ -0,0 +1,398 @@
+application that connects to the front-end dashboard using WebRTC technology,
+streams your screen, and receives video tracks.
+
+<ImageWrapper
+  url={"/img/tutorials/react-finished.gif"}
+  height={500}
+  alt="Finished app"
+/>
+
+You can check out the finished project [here](https://github.com/jellyfish-dev/jellyfish-clients-tutorials/tree/main/JellyfishDashboardReact/JellyfishDashboard)
+
+## What do you need
+
+- a little bit of experience in creating React apps
+- IDE of your choice (for example [Visual Studio Code](https://code.visualstudio.com/))
+- [Node.js](https://nodejs.org/en/) installed on your machine
+
+## Jellyfish architecture
+
+<JellyfishArchitecture />
+
+## Setup
+
+### Create React + Vite project
+
+Firstly create a brand new project.
+
+<Tabs>
+  <TabItem value="vite" label="Vite">
+
+```bash
+npm create vite@latest my-react-app -- --template react-ts
+```
+
+  </TabItem>
+</Tabs>
+
+### Add dependencies
+
+<Tabs>
+
+  <TabItem value="react" label="React">
+    For this module to work you'll need to add our `react-client-sdk` package. This
+    is necessary to create and connect Jellyfish Client.
+
+```bash npm2yarn
+npm install https://github.com/jellyfish-dev/react-client-sdk#0.1.2
+```
+
+  </TabItem>
+</Tabs>
+
+### Start the Jellyfish backend
+
+<StartingJellyfishBackend />
+
+### Start the dashboard web front-end
+
+<StartingJellyfishDashboard />
+
+## (Optional) Add a bit of CSS styling
+
+For this project, we prepared simple [CSS classes](https://github.com/Crackhoff/jellyfish-clients-tutorials/blob/main/JellyfishDashboardReact/JellyfishDashboard/src/styles.css),
+You are free to use it or create your own.
+
+## General project structure
+
+Our app will consist of two parts:
+
+- a component that will connect to the server and join the room
+
+- a component that will display the video tracks from other participants
+
+## First step - prepare all the hooks and the context
+
+To connect to the Jellyfish backend, we need to create a Membrane Client
+instance. We can do it by using the `create` function from the
+`@jellyfish-dev/react-client-sdk` package. It needs two generic parameters:
+
+- `PeerMetadata` - the type of metadata that will be sent to the server when
+  connecting to the room (for example, user name) it has to be serializable
+
+- `TrackMetadata` - the type of the metadata that will be sent to the server when
+  sending a track (for example, track name) it has to be serializable as well
+
+```tsx title="App.tsx"
+import React from "react";
+import { create } from "@jellyfish-dev/react-client-sdk";
+
+// Example metadata types for peer and track
+// You can define your metadata types just make sure they are serializable
+type PeerMetadata = {
+  name: string;
+};
+
+type TrackMetadata = {
+  type: "camera" | "screen";
+};
+
+// Create a Jellyfish client instance
+// Since we will use this context outside of the component we need to export it
+export const {
```

Comment:
We talked about moving the context provider to a separate file. Should we do it for the minimal example too? On the other hand, it makes the project more complex, which is contrary to "minimal" in the name

Reply:
I agree with the second part of your comment. Another file means another paragraph